pubsub/pubsub.go

```diff
@@ -532,6 +534,20 @@ func (s *Subscription) Receive(ctx context.Context) (_ *Message, err error) {
 			m := s.q[0]
 			s.q = s.q[1:]
 			s.throughputCount++
+			if s.ackFunc == nil {
+				// Add a finalizer that complains if the Message we return isn't
+				// acked or nacked.
+				_, file, lineno, ok := runtime.Caller(1)
+				runtime.SetFinalizer(m, func(m *Message) {
+					if !m.isAcked {
```

Comment:
I'm uncomfortable referencing this without the lock. I know, `m` is dead here, it can't possibly have a concurrent writer. Nevertheless. The rule is to lock for all reads and writes, even if you think something is impossible.

Reply:
Done.